homeassistant/components/splunk/__init__.py

```diff
@@ -55,45 +55,47 @@
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the Splunk component."""
-    conf = config[DOMAIN]
-    host = conf.get(CONF_HOST)
-    port = conf.get(CONF_PORT)
-    token = conf.get(CONF_TOKEN)
-    use_ssl = conf[CONF_SSL]
-    verify_ssl = conf.get(CONF_VERIFY_SSL)
-    name = conf.get(CONF_NAME)
-    entity_filter = conf[CONF_FILTER]
-
-    event_collector = hass_splunk(
+    """Old setup for Splunk component."""
+    if DOMAIN in config:
+        # Entity filters are not configurable in Config Flow, so are removed
+        del config[DOMAIN]["filter"]
+        _LOGGER.warning(
```

Comment:
Probably have two different messages, one for configurations with a filter and one for configurations without filter.

We should also create a repair issue here. That's required. The log is optional.

Reply:
Ive made a first pass at adding the repair issue, but I'll do more testing to ensure i did it right and cover all relevant paths.